lib/puppet/parser/type_loader.rb

```diff
@@ -113,7 +113,7 @@ def load_files(modname, files)
       # for information and it should not abort.
       # There is currently one user in indirector/resourcetype/parser
       #
-      if Puppet.lookup(:squelch_parse_errors) { || false }
+      if Puppet.lookup(:squelch_parse_errors) { false }
```

Comment:
The `||` was super confusing. It looks like a boolean OR, but I think it's saying the block receives no arguments, which is a bit clearer (though still odd) when written as:

```
Puppet.lookup(:squelch_parse_errors) do | |
 false
end
```

Note there are also usages in our YARD docs which also seem confusing, like https://github.com/puppetlabs/puppet/blob/51271eee0dcca8880b972578ef1dc34833622d22/lib/puppet/type.rb#L1992 

I think it'd be clearer to just say:

```
autorequire(:file) { ['foo', 'bar'] }
```

There's more of that in:

```
$ git grep '{||'  lib
lib/puppet/interface.rb:    # @overload define(name, version, {|| ... })
lib/puppet/interface/action_manager.rb:  # @overload action(name, {|| block})
lib/puppet/parameter.rb:    # @overload defaultto({|| ... })
lib/puppet/parameter.rb:    # @overload munge {|| ... }
lib/puppet/parameter.rb:    # @overload unmunge {|| ... }
lib/puppet/parameter.rb:    # @overload validate {|| ... }
lib/puppet/provider.rb:  # @comment a yield [ ] produces {|| ...} in the signature, do not remove the space.
lib/puppet/type.rb:  # @overload ensurable({|| ... })
lib/puppet/type.rb:  #   autorequire( 'file', {|| ['foo', 'bar'] })
lib/puppet/type.rb:  #   autobefore( 'file', {|| ['foo', 'bar'] })
lib/puppet/type.rb:  #   autosubscribe( 'file', {|| ['foo', 'bar'] })
lib/puppet/type.rb:  #   autonotify( 'file', {|| ['foo', 'bar'] })
```

Reply:
These comments should be addressed in https://github.com/puppetlabs/puppet/pull/9238/commits/821a7c4c67b31c6b0c547e2ded562bb72d3e432e.